lib/src/main/java/graphql/nadel/validation/NadelHydrationValidation.kt

```diff
@@ -227,14 +239,38 @@ internal class NadelHydrationValidation(
                 if (argument == null) {
                     MissingHydrationArgumentValueSource(parent, overallField, remoteArgSource)
                 } else {
-                    // TODO: check argument type is correct
-                    null
+                    //check the input types match with hydration and actor fields
+                    val hydrationArgType = argument.type
+                    return validateHydrationInputArg(
+                            hydrationArgType,
+                            actorFieldArg.type,
+                            parent,
+                            overallField,
+                            remoteArgDef,
+                            hydration,
+                            isBatchHydration
+                    )
                 }
             }
 
-            else -> {
-                null
+            StaticArgument -> {
+                val staticArg = remoteArgSource.staticValue
```

Comment:
wondering if this logic should be moved to `NadelHydrationArgumentValidation` like the other cases

Reply:
on second thought, this might be fine as its just a single if statement